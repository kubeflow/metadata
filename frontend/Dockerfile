FROM node:12.3.1 as build

ARG COMMIT_HASH
ARG DATE

WORKDIR /frontend

COPY . .

RUN npm install && npm rebuild node-sass && \
  npm run postinstall && \
  CI=true npm run test:coverage && \
  npm run build

# Write commit and build date files and generate the dependency licenses files
# (one for the UI and one for the webserver), concatenate them to one file
# under ./src/server
RUN mkdir -p ./server/dist && \
  echo ${COMMIT_HASH} > ./server/dist/COMMIT_HASH && \
  echo ${DATE} > ./server/dist/BUILD_DATE && \
  node gen_licenses . && \
  node gen_licenses server && \
  cat dependency-licenses.txt >> server/dependency-licenses.txt && \
  npm prune --production

FROM node:12.3.1-alpine

COPY --from=build /frontend/server /server
COPY --from=build /frontend/build /client

WORKDIR /server

EXPOSE 3000
RUN npm run build && npm prune --production
ENV API_SERVER_ADDRESS http://localhost:8080
CMD node dist/server.js ../client/ 3000
