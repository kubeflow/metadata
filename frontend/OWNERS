approvers:
  - Bobgy
  - prodonjs
  - avdaredevil
  - kwasi
reviewers:
  - Bobgy
