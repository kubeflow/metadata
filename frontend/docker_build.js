/**
 * Docker build cross platform script for KF Metadata
 * This will calculate your current hash, and build date and use them
 * to build the docker image in the format:
 * 
 * gcr.io/kubeflow-images-public/metadata-frontend:${COMMIT_HASH}
 */
const {exec, spawn} = require('child_process')
const runCmd = cmd => new Promise((res, rej) => exec(cmd, (e, out, err) => e?rej(e):res({out, err})))

// COMMIT_HASH=`git rev-parse HEAD`; docker build -t gcr.io/kubeflow-images-public/metadata-frontend:${COMMIT_HASH} --build-arg COMMIT_HASH=${COMMIT_HASH} --build-arg DATE=\"`date -u`\" .
const main = async _ => {
    const commitHash = (await runCmd('git rev-parse HEAD')).out.trim()
    const date = new Date().toUTCString()
    const protocProcess = spawn(
        'docker', [
            `build`,
            `-t`,`gcr.io/kubeflow-images-public/metadata-frontend:${commitHash}`,
            `--build-arg`,`COMMIT_HASH=${commitHash}`,
            `--build-arg`,`DATE="${date}"`,
            `.`
        ], {shell: true}
    );
    protocProcess.stdout.on('data', buffer => console.log(buffer.toString()))
    protocProcess.stderr.on('data', buffer => console.error(buffer.toString()))
    protocProcess.on('close', code => {
        if (code) return
        console.log(`Docker image built!`)
    })
}

main()
