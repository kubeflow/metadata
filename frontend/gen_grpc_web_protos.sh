#!/usr/bin/env bash
PROTOC_GEN_TS_PATH="./node_modules/.bin/protoc-gen-ts"
OUT_DIR="./src/generated"

if [[ ! -d ${OUT_DIR} ]]; then
  mkdir ${OUT_DIR}
fi

# Expects protoc to be on your PATH.
#
# From npm/google-protobuf:
# The compiler is not currently available via npm, but you can download a
# pre-built binary on GitHub (look for the protoc-*.zip files under Downloads).
protoc \
    --plugin="protoc-gen-ts=${PROTOC_GEN_TS_PATH}" \
    --js_out="import_style=commonjs,binary:${OUT_DIR}" \
    --ts_out="service=grpc-web:${OUT_DIR}" \
    src/apis/**/*.proto
