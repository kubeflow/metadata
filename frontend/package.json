{
  "name": "metadata-frontend",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^3.7.1",
    "@material-ui/icons": "^3.0.1",
    "grpc-web": "^1.0.7",
    "lodash.debounce": "^4.0.8",
    "lodash.groupby": "^4.6.0",
    "lodash.isfunction": "^3.0.9",
    "node-sass": "^4.13.0",
    "portable-fetch": "^3.0.0",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-router-dom": "^4.3.1",
    "react-scripts": "^3.3.0",
    "react-svg-line-chart": "^2.0.2",
    "typestyle": "^2.0.1"
  },
  "devDependencies": {
    "@types/enzyme": "^3.9.3",
    "@types/enzyme-adapter-react-16": "^1.0.5",
    "@types/express": "^4.16.0",
    "@types/google-protobuf": "^3.7.2",
    "@types/jest": "^24.0.13",
    "@types/lodash.debounce": "^4.0.6",
    "@types/lodash.groupby": "^4.6.6",
    "@types/lodash.isfunction": "^3.0.6",
    "@types/node": "^12.0.2",
    "@types/react": "^16.8.18",
    "@types/react-dom": "^16.8.4",
    "@types/react-router-dom": "^4.3.1",
    "cross-env": "^6.0.3",
    "enzyme": "^3.10.0",
    "enzyme-adapter-react-16": "^1.14.0",
    "enzyme-to-json": "^3.3.5",
    "eslint-plugin-import": "^2.18.2",
    "license-checker": "^25.0.1",
    "react-router-test-context": "^0.1.0",
    "ts-protoc-gen": "^0.12.0",
    "typescript": "^3.4.5"
  },
  "scripts": {
    "apis": "npm run apis:api && npm run apis:service",
    "apis:api": "java -jar swagger-codegen-cli.jar generate -i ../api/api.swagger.json -l typescript-fetch -o ./src/apis/api -c ./swagger-config.json",
    "apis:metadata": "node ./gen_grpc_web_protos.js",
    "apis:service": "java -jar swagger-codegen-cli.jar generate -i ../api/service.swagger.json -l typescript-fetch -o ./src/apis/service -c ./swagger-config.json",
    "build": "cross-env EXTEND_ESLINT=true react-scripts build",
    "docker": "node ./docker_build.js",
    "eject": "cross-env EXTEND_ESLINT=true react-scripts eject",
    "postinstall": "cd ./server && npm i",
    "start": "cross-env EXTEND_ESLINT=true react-scripts start",
    "start:proxy": "node ./start-proxy.js",
    "test": "cross-env TZ='America/New_York' EXTEND_ESLINT=true react-scripts test --env=jsdom",
    "test:coverage": "npm test -- --coverage --watchAll=false"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "eslintIgnore": [
    "src/__mocks__",
    "src/generated"
  ],
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "homepage": "./",
  "jest": {
    "collectCoverageFrom": [
      "src/**/*.{ts,tsx}",
      "!src/__mocks__",
      "!src/apis/**/*.{ts,tsx}",
      "!src/generated/**/*",
      "!<rootDir>/node_modules/"
    ],
    "coverageReporters": [
      "text",
      "lcov"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ]
  },
  "proxy": "http://localhost:8080"
}
