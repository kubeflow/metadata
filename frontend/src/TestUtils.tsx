/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* eslint-disable */
// tslint:disable: object-literal-sort-keys

// Because this is test utils.

import * as React from 'react';
// @ts-ignore
import createRouterContext from 'react-router-test-context';
import {PageProps, Page} from './pages/Page';
import {ToolbarActionConfig} from './components/Toolbar';
import {match} from 'react-router';
import {mount, ReactWrapper} from 'enzyme';
import {object} from 'prop-types';
import {format} from 'prettier';
import snapshotDiff from 'snapshot-diff';
import {
  Artifact,
  ArtifactProperties,
  ArtifactCustomProperties,
  Value,
} from '@kubeflow/frontend';

export default class TestUtils {
  /**
   * Mounts the given component with a fake router and returns the mounted tree
   */
  // tslint:disable-next-line:variable-name
  public static mountWithRouter(component: React.ReactElement<any>): ReactWrapper {
    const childContextTypes = {
      router: object,
    };
    const context = createRouterContext();
    const tree = mount(component, { context, childContextTypes });
    return tree;
  }

  /**
   * Flushes all already queued promises and returns a promise. Note this will
   * only work if the promises have already been queued, so it cannot be used to
   * wait on a promise that hasn't been dispatched yet.
   */
  public static flushPromises(): Promise<void> {
    return new Promise(resolve => setImmediate(resolve));
  }

  /**
   * Adds a one-time mock implementation to the provided spy that mimics an error
   * network response
   */
  public static makeErrorResponseOnce(spy: jest.MockInstance<unknown, unknown[]>, message: string): void {
    spy.mockImplementationOnce(() => {
      throw {
        text: () => Promise.resolve(message),
      };
    });
  }

  /**
   * Generates a customizable PageProps object that can be passed to initialize
   * Page components, taking care of setting ToolbarProps properly, which have
   * to be set after component initialization.
   */
  // tslint:disable-next-line:variable-name
  public static generatePageProps(
    PageElement: new (_: PageProps) => Page<any, any>,
    location: Location,
    matchValue: match,
    historyPushSpy: jest.SpyInstance<unknown> | null,
    updateBannerSpy: jest.SpyInstance<unknown> | null,
    updateDialogSpy: jest.SpyInstance<unknown> | null,
    updateToolbarSpy: jest.SpyInstance<unknown> | null,
    updateSnackbarSpy: jest.SpyInstance<unknown> | null,
  ): PageProps {
    const pageProps = {
      history: { push: historyPushSpy } as any,
      location: location as any,
      match: matchValue,
      toolbarProps: { actions: {}, breadcrumbs: [], pageTitle: '' },
      updateBanner: updateBannerSpy as any,
      updateDialog: updateDialogSpy as any,
      updateSnackbar: updateSnackbarSpy as any,
      updateToolbar: updateToolbarSpy as any,
    } as PageProps;
    pageProps.toolbarProps = new PageElement(pageProps).getInitialToolbarState();
    // The toolbar spy gets called in the getInitialToolbarState method, reset it
    // in order to simplify tests
    if (updateToolbarSpy) {
      updateToolbarSpy.mockReset();
    }
    return pageProps;
  }

  public static getToolbarButton(
    updateToolbarSpy: jest.SpyInstance<unknown>,
    buttonKey: string,
  ): ToolbarActionConfig {
    const lastCallIdx = updateToolbarSpy.mock.calls.length - 1;
    const lastCall = updateToolbarSpy.mock.calls[lastCallIdx][0];
    return lastCall.actions[buttonKey];
  }
}

/**
 * Generate diff text for two HTML strings.
 * Recommend providing base and update annotations to clarify context in the diff directly.
 */
export function diffHTML({
  base,
  update,
  baseAnnotation,
  updateAnnotation,
}: {
  base: string;
  baseAnnotation?: string;
  update: string;
  updateAnnotation?: string;
}) {
  return diff({
    base: formatHTML(base),
    update: formatHTML(update),
    baseAnnotation,
    updateAnnotation,
  });
}

export function diff({
  base,
  update,
  baseAnnotation,
  updateAnnotation,
}: {
  base: string;
  baseAnnotation?: string;
  update: string;
  updateAnnotation?: string;
}) {
  return snapshotDiff(base, update, {
    stablePatchmarks: true, // Avoid line numbers in diff, so that diffs are stable against irrelevant changes
    aAnnotation: baseAnnotation,
    bAnnotation: updateAnnotation,
  });
}

function formatHTML(html: string): string {
  return format(html, { parser: 'html' });
}

/**	
 * Generates a customizable PageProps object that can be passed to initialize	
 * Page components, taking care of setting ToolbarProps properly, which have	
 * to be set after component initialization.	
 */	
// tslint:disable-next-line:variable-name	
export function generatePageProps(	
  // tslint:disable-next-line: variable-name
  PageElement: new (_: PageProps) => Page<any, any>, location: Location,	
  matchValue: match, historyPushSpy: jest.SpyInstance | null,	
  updateBannerSpy: jest.SpyInstance | null,	
  updateDialogSpy: jest.SpyInstance | null,	
  updateToolbarSpy: jest.SpyInstance | null,	
  updateSnackbarSpy: jest.SpyInstance | null): PageProps {	
  const pageProps = {	
    history: {push: historyPushSpy} as any,	
    location: location as any,	
    match: matchValue,	
    toolbarProps: {actions: [], breadcrumbs: [], pageTitle: ''},	
    updateBanner: updateBannerSpy as any,	
    updateDialog: updateDialogSpy as any,	
    updateSnackbar: updateSnackbarSpy as any,	
    updateToolbar: updateToolbarSpy as any,	
  } as unknown as PageProps;	
  pageProps.toolbarProps = new PageElement(pageProps).getInitialToolbarState();	
  // The toolbar spy gets called in the getInitialToolbarState method, reset	
  // it in order to simplify tests	
  if (updateToolbarSpy) {	
    updateToolbarSpy.mockReset();	
  }	
  return pageProps;	
}	

export function getToolbarButton(	
  updateToolbarSpy: jest.SpyInstance, title: string): ToolbarActionConfig {	
  const lastCallIdx = updateToolbarSpy.mock.calls.length - 1;	
  const lastCall = updateToolbarSpy.mock.calls[lastCallIdx][0];	
  return lastCall.actions.find((b: any) => b.title === title);	
}	

export const doubleValue = (number: number) => {	
  const value = new Value();	
  value.setDoubleValue(number);	
  return value;	
};	

export const intValue = (number: number) => {	
  const value = new Value();	
  value.setIntValue(number);	
  return value;	
};	

export const stringValue = (string: String) => {	
  const value = new Value();	
  value.setStringValue(String(string));	
  return value;	
};	

export const buildTestModel = () => {	
  const model = new Artifact();	
  model.setId(1);	
  model.setTypeId(1);	
  model.setUri('gs://my-bucket/mnist');	
  model.getPropertiesMap().set(ArtifactProperties.NAME, stringValue('test model'));	
  model.getPropertiesMap().set(ArtifactProperties.DESCRIPTION, stringValue('A really great model'));	
  model.getPropertiesMap().set(ArtifactProperties.VERSION, stringValue('v1'));	
  model.getPropertiesMap().set(ArtifactProperties.CREATE_TIME, stringValue('2019-06-12T01:21:48.259263Z'));	
  model.getPropertiesMap().set(ArtifactProperties.ALL_META, stringValue(	
      '{"hyperparameters": {"early_stop": true, ' +	
      '"layers": [10, 3, 1], "learning_rate": 0.5}, ' +	
      '"model_type": "neural network", ' +	
      '"training_framework": {"name": "tensorflow", "version": "v1.0"}}'));	
  model.getCustomPropertiesMap().set(ArtifactCustomProperties.WORKSPACE, stringValue('workspace-1'));	
  model.getCustomPropertiesMap().set(ArtifactCustomProperties.RUN, stringValue('1'));	
  return model	
};	

export const testModel = buildTestModel();