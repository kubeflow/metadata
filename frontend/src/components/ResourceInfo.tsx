/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from 'react';
import {stylesheet} from 'typestyle';
import {color, commonCss} from '../Css';
import {MlMetadataArtifact, MlMetadataExecution} from '../apis/service';
import {getMlMetadataMetadataValue} from '../lib/Utils';

export const css = stylesheet({
  resourceInfo: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  field: {
    flexBasis: '300px',
    marginBottom: '32px',
  },
  term: {
    color: color.grey,
    fontSize: '12px',
    letterSpacing: '0.2px',
    lineHeight: '16px',
  },
  value: {
    color: color.secondaryText,
    fontSize: '14px',
    letterSpacing: '0.2px',
    lineHeight: '20px',
  }
});

export interface ResourceInfoProps {
  typeName: string;
  resource: MlMetadataArtifact | MlMetadataExecution;
}

export class ResourceInfo extends React.Component<ResourceInfoProps, {}> {

  public render(): JSX.Element {
    const { resource } = this.props;
    return (
      <section>
        <h1 className={commonCss.header}>Type: {this.props.typeName}</h1>
        <h2 className={commonCss.header2}>Properties</h2>
        <dl className={css.resourceInfo}>
          {Object.keys(resource.properties || {})
            // TODO: __ALL_META__ is something of a hack, is redundant, and can be ignored
            .filter(k => k !== '__ALL_META__')
            .map(k =>
              <div className={css.field} key={k}>
                <dt className={css.term}>{k}</dt>
                <dd className={css.value}>
                  {resource.properties && getMlMetadataMetadataValue(resource.properties[k])}
                </dd>
              </div>
            )
          }
        </dl>
        <h2 className={commonCss.header2}>Custom Properties</h2>
        <dl className={css.resourceInfo}>
          {Object.keys(resource.custom_properties || {}).map(k =>
            <div className={css.field} key={k}>
              <dt className={css.term}>{k}</dt>
              <dd className={css.value}>
                {resource.custom_properties && getMlMetadataMetadataValue(resource.custom_properties[k])}
              </dd>
            </div>
          )}
        </dl>
      </section>
    );
  }
}
