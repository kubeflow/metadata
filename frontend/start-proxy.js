const {promisify} = require('util');
const child_process = require('child_process');

const exec = promisify(child_process.exec);
const spawn = child_process.spawn;

const NAMESPACE = process.env['NAMESPACE'] ? process.env['NAMESPACE'] : 'kubeflow';
const KUBECTL = 'kubectl';

console.log(`
===============================================================================
This script helps set up a dev env for client side UI only. It uses a real KF deployment for api requests.

What this does:
1. It detects metadata ui pod name in a KF deployment.
2. Port forward metadata ui pod to localhost:8080 (Metadata UI dev env is configured to redirect all api requests to localhost:8080)
===============================================================================
`);

const detectMetadataUiPodName = async () => {
  console.log('Detecting metadata ui pod name...');

  const args = [
    'get',
    'pods',
    `-n ${NAMESPACE}`,
    `-l app=metadata-ui`,
    `-o=custom-columns=:.metadata.name`,
    `--no-headers`
  ];

  const command = `${KUBECTL} ${args.join(' ')}`;

  const {stdout, stderr} = await exec(command);

  if (stderr.length > 0) {
    console.log(`Error running kubectl:`);
    console.log(`Command: ${command}`);
    console.log(`Error: ${stderr}`);
  }

  if (stdout.length === 0) {
    console.log(`Couldn't get metadata-ui pod in namespace '$NAMESPACE', check the cluster your ` +
        `kubectl talks to and your namespace is correct.`);
    console.log(`Namespace can be configured by setting env variable NAMESPACE. e.g. ` +
        `'$ NAMESPACE=kfm npm run start:proxy'`);
    process.exit(1);
  }

  const podName = stdout.trim();
  console.log(`Metadata UI pod is ${podName}`);

  return podName
};

const portForwardToMetadataUiPod = async (podName) => {
  console.log('Starting to port forward frontend server in a KF deployment to respond to apis...');

  const args = [
    'port-forward',
    `-n`,
    `${NAMESPACE}`,
    podName,
    '8080:3000'
  ];

  const logBuffer = (data) => {
    console.log(data.toString().trim());
  };

  const proxyProcess = spawn(KUBECTL, args);
  proxyProcess.stdout.on('data', logBuffer);
  proxyProcess.stderr.on('data', logBuffer);
  proxyProcess.on('close', code => {
    console.log(`Metadata UI port forwarding process exited with code ${code}.`)
  });
};

detectMetadataUiPodName()
    .then(podName => portForwardToMetadataUiPod(podName))
    .catch(reason => {
      console.log(reason);
    });
