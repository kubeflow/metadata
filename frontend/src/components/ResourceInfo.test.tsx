import {Artifact, ArtifactCustomProperties, ArtifactProperties} from 'frontend';
import * as React from 'react';
import {ResourceInfo} from './ResourceInfo';
import {shallow, ShallowWrapper} from 'enzyme';
import {stringValue, testModel} from '../TestUtils';

describe('ResourceInfo', () => {
  let tree: ShallowWrapper;

  afterEach(() => tree.unmount());

  it('Renders generic Artifact information', () => {
    const dataset = new Artifact();
    dataset.setId(2);
    dataset.setTypeId(3);
    dataset.setUri('gs://my-bucket/dataset');
    dataset.getPropertiesMap().set(ArtifactProperties.NAME, stringValue('test dataset'));
    dataset.getPropertiesMap().set(ArtifactProperties.VERSION, stringValue('v0.0.1'));
    dataset.getPropertiesMap().set(ArtifactProperties.CREATE_TIME, stringValue('2019-06-12T01:21:48.259263Z'));
    dataset.getCustomPropertiesMap().set(ArtifactCustomProperties.WORKSPACE, stringValue('workspace-1'));

    tree = shallow(<ResourceInfo typeName='Data Set' resource={dataset} />);

    expect(tree).toMatchSnapshot();
  });

  it('Renders Artifact information for a Model', () => {
    tree = shallow(<ResourceInfo typeName='Model' resource={testModel} />);
    expect(tree).toMatchSnapshot();
  });
});
