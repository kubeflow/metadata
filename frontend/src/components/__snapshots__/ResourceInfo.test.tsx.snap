// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceInfo Renders Artifact information for a Model 1`] = `
<section>
  <h1
    className="header"
  >
    Type: 
    Model
  </h1>
  <h2
    className="header2"
  >
    Properties
  </h2>
  <dl
    className="resourceInfo"
  >
    <div
      className="field"
      key="create_time"
    >
      <dt
        className="term"
      >
        create_time
      </dt>
      <dd
        className="value"
      >
        2019-06-12T01:21:48.259263Z
      </dd>
    </div>
    <div
      className="field"
      key="description"
    >
      <dt
        className="term"
      >
        description
      </dt>
      <dd
        className="value"
      >
        A really great model
      </dd>
    </div>
    <div
      className="field"
      key="name"
    >
      <dt
        className="term"
      >
        name
      </dt>
      <dd
        className="value"
      >
        test model
      </dd>
    </div>
    <div
      className="field"
      key="version"
    >
      <dt
        className="term"
      >
        version
      </dt>
      <dd
        className="value"
      >
        v1
      </dd>
    </div>
  </dl>
  <h2
    className="header2"
  >
    Custom Properties
  </h2>
  <dl
    className="resourceInfo"
  >
    <div
      className="field"
      key="__kf_run__"
    >
      <dt
        className="term"
      >
        __kf_run__
      </dt>
      <dd
        className="value"
      >
        1
      </dd>
    </div>
    <div
      className="field"
      key="__kf_workspace__"
    >
      <dt
        className="term"
      >
        __kf_workspace__
      </dt>
      <dd
        className="value"
      >
        workspace-1
      </dd>
    </div>
  </dl>
</section>
`;

exports[`ResourceInfo Renders generic Artifact information 1`] = `
<section>
  <h1
    className="header"
  >
    Type: 
    Data Set
  </h1>
  <h2
    className="header2"
  >
    Properties
  </h2>
  <dl
    className="resourceInfo"
  >
    <div
      className="field"
      key="create_time"
    >
      <dt
        className="term"
      >
        create_time
      </dt>
      <dd
        className="value"
      >
        2019-06-12T01:21:48.259263Z
      </dd>
    </div>
    <div
      className="field"
      key="name"
    >
      <dt
        className="term"
      >
        name
      </dt>
      <dd
        className="value"
      >
        test dataset
      </dd>
    </div>
    <div
      className="field"
      key="version"
    >
      <dt
        className="term"
      >
        version
      </dt>
      <dd
        className="value"
      >
        v0.0.1
      </dd>
    </div>
  </dl>
  <h2
    className="header2"
  >
    Custom Properties
  </h2>
  <dl
    className="resourceInfo"
  >
    <div
      className="field"
      key="__kf_workspace__"
    >
      <dt
        className="term"
      >
        __kf_workspace__
      </dt>
      <dd
        className="value"
      >
        workspace-1
      </dd>
    </div>
  </dl>
</section>
`;
